actor/builtin/miner/miner.go

```diff
@@ -516,6 +519,25 @@ func (ma *Actor) GetPoStState(ctx exec.VMContext) (*big.Int, uint8, error) {
 	return big.NewInt(result), 0, nil
 }
 
+// IsProving returns whether this miner is proving sectors, i.e. if the miner's proving set
+// is not empty.
+func (ma *Actor) IsProving(ctx exec.VMContext) (bool, uint8, error) {
+	var state State
+	out, err := actor.WithState(ctx, &state, func() (interface{}, error) {
+		return state.ProvingSet.Size() != 0, nil
+	})
+
+	if err != nil {
+		return false, errors.CodeError(err), err
+	}
+	result, ok := out.(bool)
+	if !ok {
+		return false, 1, errors.NewFaultErrorf("expected a bool, but got %T instead", out)
+	}
```

Comment:
FYI I added some other helpers to make read-only methods less verbose. See `GetProvingPeriod` for an example.

Reply:
ah, I didn't even see loadState.  Thanks for the pointer.